src/app/component/user/components/profile/profile-service/profile.service.ts

```diff
@@ -40,14 +40,15 @@ export class ProfileService {
   }
 
   public getShoppingList(): Observable<ShoppingList[]> {
-    this.setUserId();
     const currentLang = this.languageService.getCurrentLanguage();
+    this.setUserId();
     return this.http.get<ShoppingList[]>(`
     ${mainLink}custom/shopping-list-items/${this.userId}/custom-shopping-list-items?lang=${currentLang}
     `);
   }
 
   public getUserProfileStatistics(): Observable<ProfileStatistics> {
+    this.setUserId();
```

Comment:
why do we need to call this method here ?

Reply:
because we get statistics of profile by user id